masonry/src/widget/progress_bar.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright 2019 the Xilem Authors and the Druid Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! A checkbox widget.
+
+use accesskit::Role;
+use kurbo::Point;
+use smallvec::{smallvec, SmallVec};
+use tracing::{trace, trace_span, Span};
+use vello::Scene;
+
+use crate::kurbo::Size;
+use crate::paint_scene_helpers::{fill_lin_gradient, stroke, UnitPoint};
+use crate::text::TextLayout;
+use crate::widget::WidgetMut;
+use crate::{
+    theme, AccessCtx, AccessEvent, ArcStr, BoxConstraints, EventCtx, LayoutCtx, LifeCycle,
+    LifeCycleCtx, PaintCtx, PointerEvent, StatusChange, TextEvent, Widget, WidgetId,
+};
+
+/// A checkbox that can be toggled.
+pub struct ProgressBar {
+    /// A value in the range `[0, 1]` inclusive, where 0 is 0% and 1 is 100% complete.
+    ///
+    /// `None` variant can be used to show a progress bar without a percentage.
+    /// It is also used if an invalid float (outside of [0, 1]) is passed.
+    part_complete: Option<f32>,
```

Comment:
I'd rename this (and the matching methods and function arguments) to `progress` instead.